### PR TITLE
Blaze: Laying foundations for showing Advertising in menu for Jetpack sites

### PR DESCRIPTION
**Pull request by candy02058912**
#### Changes proposed in this Pull Request:

* Utilizes the Blaze package to control whether to show Advertising in menu for Jetpack sites

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

#### Testing instructions:

* Check if unit tests pass

### CODE REVIEW

**Review by lupus2k — DISMISSED** [review]
LGTM, menu and link work as expected.

image

**Comment by sdixon194** [comment]
This works for me! Do we want to also add a test to `projects/plugins/jetpack/tests/php/modules/masterbar/test-class-admin-menu.php` to ensure the menu item exists?

**Comment by candy02058912** [comment]
> This works for me! Do we want to also add a test to `projects/plugins/jetpack/tests/php/modules/masterbar/test-class-admin-menu.php` to ensure the menu item exists?

@sdixon194 Thanks for mentioning it! I added the test to `projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php` since that's the file this PR is changing.

**Review by jeherve — COMMENTED** [review]
This works well for me, but I wonder if we should worry about users that do not have access to Blaze yet?

I was mentioning this in 1386-gh-Automattic/dotcom-forge#issuecomment-1351085107 for example. Maybe we should apply the same gating for now (language)?

**Comment by candy02058912** [comment]
> This works well for me, but I wonder if we should worry about users that do not have access to Blaze yet?
> 
> I was mentioning this in [1386-gh-Automattic/dotcom-forge#issuecomment-1351085107](https://href.li/?https://github.com/Automattic/dotcom-forge/issues/1386#issuecomment-1351085107) for example. Maybe we should apply the same gating for now (language)?

Thanks for the suggestion @jeherve! I added a locale check for the site's locale.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php [hunk 1]
```diff
@@ -231,8 +231,10 @@ public function add_users_menu() {
 	 */
 	public function add_tools_menu() {
 		add_menu_page( esc_attr__( 'Tools', 'jetpack' ), __( 'Tools', 'jetpack' ), 'publish_posts', 'tools.php', null, 'dashicons-admin-tools', 75 );
-
-		add_submenu_page( 'tools.php', esc_attr__( 'Advertising', 'jetpack' ), __( 'Advertising', 'jetpack' ), 'manage_options', 'https://wordpress.com/advertising/' . $this->domain, null, 1 );
+		// TODO: currently only open to sites that have en locale.
+		if ( false !== strpos( get_locale(), 'en_' ) ) {
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php]: I believe we're currently looking at the user locale, not the site locale. The site locale may make more sense on the long run in my opinion, but maybe we want to keep using the user locale here as well, for consistency?

I believe you can get that info from `(new Automattic\Jetpack\Connection\Manager() )->get_connected_user_data()['user_locale']`

This again highlights the need for a more accessible way to check if a **site** is eligible for Blaze, as I was mentioning here:
1386-gh-Automattic/dotcom-forge#issuecomment-1351085107

Comment (paulopmt1) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php]: > but maybe we want to keep using the user locale here as well, for consistency?

Sounds good to me @jeherve! We'll start soon working on translating Blaze so we'll be able to remove this check altogether: p1673035073317699-slack-C04DZ725FEK

Comment (candy02058912) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php]: > I believe you can get that info from (new Automattic\Jetpack\Connection\Manager() )->get_connected_user_data()['user_locale']

Thanks for mentioning this. It's super helpful 🙏

**Comment by candy02058912** [comment]
> I believe you can get that info from (new Automattic\Jetpack\Connection\Manager() )->get_connected_user_data()['user_locale']

@jeherve I've updated the code, but am finding the below behavior a bit weird:

When I was testing out this function and changing my user locale to something else in Calypso (https://wordpress.com/me/account), the value was always `en`.
I think that I am probably understanding this user locale wrongly.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php [hunk 2]
```diff
@@ -231,7 +233,11 @@ public function add_users_menu() {
 	 */
 	public function add_tools_menu() {
 		add_menu_page( esc_attr__( 'Tools', 'jetpack' ), __( 'Tools', 'jetpack' ), 'publish_posts', 'tools.php', null, 'dashicons-admin-tools', 75 );
-
+		// TODO: remove locale restriction when blaze is open to everyone.
+		$user_locale = ( new Manager() )->get_connected_user_data()['user_locale'];
+		if ( 'en' == $user_locale ) {
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php]: Maybe we could do it like that instead, to handle both en and en-gb, like on WordPress.com?

```suggestion
		$user_data = ( new Manager() )->get_connected_user_data();
		if (
			! empty( $user_data['user_locale'] )
			&& in_array( $user_data['user_locale'], array( 'en', 'en-gb' ), true )
		) {
```

**Review by jeherve — CHANGES_REQUESTED** [review]
> When I was testing out this function and changing my user locale to something else in Calypso ([wordpress.com/me/account](https://wordpress.com/me/account)), the value was always `en`.

The data is cached for a day, so it can be a bit more difficult to test. You would have to delete the transient starting with `jetpack_connected_user_data_`, and then visit the Jetpack dashboard in wp-admin to refresh the data, and that should work.

It looks like you'll need to update the test following the change though.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php (hunk 2 above)
Comment (candy02058912) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php]: >  You would have to delete the transient starting with jetpack_connected_user_data_, and then visit the Jetpack dashboard in wp-admin to refresh the data, and that should work.

Yes, it works. Learned about transients today 🙏 
I've also updated the unit tests so they're all good now!

**Review by jeherve — COMMENTED** [review]
Does this work well on your end? I'm getting the following notices on my end:

```
[10-Jan-2023 14:34:20 UTC] PHP Notice:  Undefined property: Automattic\Jetpack\Dashboard_Customizations\Jetpack_Admin_Menu::$connection in wp-content/plugins/jetpack-dev/modules/masterbar/admin-menu/class-jetpack-admin-menu.php on line 254
[10-Jan-2023 14:34:20 UTC] PHP Fatal error:  Uncaught Error: Call to a member function get_connected_user_data() on null in wp-content/plugins/jetpack-dev/modules/masterbar/admin-menu/class-jetpack-admin-menu.php:254
```

**Comment by candy02058912** [comment]
Updated, I was too focused on getting the unit tests to pass yesterday 😞
Checked:
- [x] Unit test `jetpack docker phpunit -- --filter=Test_Jetpack_Admin_Menu` 
- [x] synced them to my jetpack site
- [x] tested on wordpress.com and checked the debug.log.
- [x] tested only `en` and `en-gb` shows Advertising in the Tools submenu

Thank you for your patience 🙏

**Comment by jeherve** [comment]
Coming back to this, and as I'm working on tweaking the acceptance criteria for Blaze (see #28266 and #28267), I was wondering if we should start leveraging this instead? Something like 

```diff
diff --git a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php
index 1229ab13e5..6c2ff6faaf 100644
--- a/projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php
+++ b/projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php
@@ -7,7 +7,7 @@
 
 namespace Automattic\Jetpack\Dashboard_Customizations;
 
-use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Blaze;
 
 require_once __DIR__ . '/class-admin-menu.php';
 
@@ -15,19 +15,6 @@ require_once __DIR__ . '/class-admin-menu.php';
  * Class Jetpack_Admin_Menu.
  */
 class Jetpack_Admin_Menu extends Admin_Menu {
-
-	/**
-	 * Jetpack_Admin_Menu constructor.
-	 * Currently used to initialize the connection manager for getting the user locale
-	 * for showing the Advertising menu for add_tools_menu.
-	 * TODO: removed after Blaze has been released to all locales.
-	 */
-	protected function __construct() {
-		parent::__construct();
-
-		$this->connection = new Connection_Manager();
-	}
-
 	/**
 	 * Determines whether the current locale is right-to-left (RTL).
 	 *
@@ -246,11 +233,8 @@ class Jetpack_Admin_Menu extends Admin_Menu {
 	public function add_tools_menu() {
 		add_menu_page( esc_attr__( 'Tools', 'jetpack' ), __( 'Tools', 'jetpack' ), 'publish_posts', 'tools.php', null, 'dashicons-admin-tools', 75 );
 		// TODO: remove locale restriction when blaze is open to everyone.
-		$user_data = $this->connection->get_connected_user_data();
-		if (
-			! empty( $user_data['user_locale'] )
-			&& in_array( $user_data['user_locale'], array( 'en', 'en-gb' ), true )
-		) {
+
+		if ( Blaze::should_initialize() ) {
 			add_submenu_page( 'tools.php', esc_attr__( 'Advertising', 'jetpack' ), __( 'Advertising', 'jetpack' ), 'manage_options', 'https://wordpress.com/advertising/' . $this->domain, null, 1 );
 		}
 		add_submenu_page( 'tools.php', esc_attr__( 'Marketing', 'jetpack' ), __( 'Marketing', 'jetpack' ), 'publish_posts', 'https://wordpress.com/marketing/tools/' . $this->domain );
diff --git a/projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php b/projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php
index 264bd3fb7b..592ff763e1 100644
--- a/projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php
+++ b/projects/plugins/jetpack/tests/php/modules/masterbar/test-class-jetpack-admin-menu.php
@@ -5,7 +5,6 @@
  * @package automattic/jetpack
  */
 
-use Automattic\Jetpack\Connection\Manager as Connection_Manager;
 use Automattic\Jetpack\Dashboard_Customizations\Jetpack_Admin_Menu;
 use Automattic\Jetpack\Status;
 
@@ -111,14 +110,8 @@ class Test_Jetpack_Admin_Menu extends WP_UnitTestCase {
 	public function test_add_tools_menu() {
 		global $submenu;
 
-		// TODO: $connection is used to get locale data for the initial restriction.
-		// Remove after all locale restrictions have been lifted.
-		$connection = $this->createMock( Connection_Manager::class );
-		$user_data  = array(
-			'user_locale' => 'en',
-		);
-		$connection->method( 'get_connected_user_data' )->willReturn( $user_data );
-		static::$admin_menu->connection = $connection;
+		// This filter should override everything, even if it's not connected.
+		add_filter( 'jetpack_blaze_enabled', '__return_true' );
 
 		static::$admin_menu->add_tools_menu();
 
```

What do you think?

**Comment by candy02058912** [comment]
I've updated the PR and changed the description to reflect what this PR is actually doing.
Thanks @jeherve

**Review by jeherve — APPROVED** [review]

**Pull request merged**